tests/cpu/compare_with_acts_seeding.cpp

```diff
@@ -247,6 +247,33 @@ TEST_P(CompareWithActsSeedingTests, Run) {
     // create grid with bin sizes according to the configured geometry
     std::unique_ptr<Acts::SpacePointGrid<SpacePoint>> grid =
         Acts::SpacePointGridCreator::createGrid<SpacePoint>(gridConf);
+
+    const auto numLocalBins = grid->numLocalBins();
+    // Currently traccc is using grid2. check if acts grid dimensionality is the
+    // same
+    EXPECT_EQ(numLocalBins.size(), 2);
+
+    detray::axis::circular axis0 = internal_spacepoints_per_event.axis_p0();
+    detray::axis::regular axis1 = internal_spacepoints_per_event.axis_p1();
+
+    auto acts_axis0 = grid->axes()[0];
+    auto acts_axis1 = grid->axes()[1];
+
+    EXPECT_EQ(axis0.bins(), acts_axis0->getNBins());
+    EXPECT_EQ(axis1.bins(), acts_axis1->getNBins());
+
+    auto axis0_borders = axis0.all_borders();
+    auto axis1_borders = axis1.all_borders();
+    auto acts_axis0_borders = acts_axis0->getBinEdges();
+    auto acts_axis1_borders = acts_axis1->getBinEdges();
+
```

Comment:
It might be a bit too verbose, but I think it would make sense to add the following tests as well:

```
EXPECT_EQ(axis0_borders.size(), axis0.bins() + 1);
EXPECT_EQ(axis0_borders.size(), acts_axis0_borders.size());

EXPECT_EQ(axis1_borders.size(), axis1.bins() + 1);
EXPECT_EQ(axis1_borders.size(), acts_axis1_borders.size());
```

So that the assumptions used in the tests in lines 270-275 would be explicitly tested.

Reply:
Is it not safe to assume that the size of the borders vector is = #bins + 1?

If so, these tests are redundant as there's already one which tests that #bins is the same.

` EXPECT_EQ(axis0.bins(), acts_axis0->getNBins());`
` EXPECT_EQ(axis1.bins(), acts_axis1->getNBins());`

Reply:
That is just how the grids are supposed to work. The goal of a unit test is exactly to test these sort of things. As far as this test is concerned, the objects returned for the bin borders have nothing to do with the values returned bu the `bins()` and `getNBins()` calls.